backend/building-block/src/main/kotlin/com/ritense/buildingblock/service/BuildingBlockDefinitionArtworkService.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2015-2025 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.buildingblock.service
+
+import com.ritense.buildingblock.domain.definition.BuildingBlockDefinitionArtwork
+import com.ritense.buildingblock.exception.UnknownBuildingBlockDefinitionException
+import com.ritense.buildingblock.repository.BuildingBlockDefinitionArtworkRepository
+import com.ritense.buildingblock.repository.BuildingBlockDefinitionRepository
+import com.ritense.buildingblock.web.rest.dto.BuildingBlockDefinitionArtworkDto
+import com.ritense.buildingblock.web.rest.dto.CreateBuildingBlockDefinitionArtworkDto
+import com.ritense.valtimo.contract.buildingblock.BuildingBlockDefinitionId
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.awt.Graphics2D
+import java.awt.RenderingHints
+import java.awt.image.BufferedImage
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.util.Base64
+import javax.imageio.ImageIO
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+@Service
+class BuildingBlockDefinitionArtworkService(
```

Comment:
All service calls should deny access control I think, and then on places where these methods are called, use runWithoutAuthorization